### PR TITLE
[Related #48] Initialize websockets

### PR DESCRIPTION
**Pull request by supunTE**
Related #48 
Initialized web sockets on the client and the server.

### CODE REVIEW

File: apps/client/src/services/initialize-socket.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { io, Socket } from 'socket.io-client';
+
+const SOCKET_ENDPOINT =
+  process.env['NX_SOCKET_ENDPOINT'] || 'http://localhost:3000';
```
Comment (dalanad) [inline apps/client/src/services/initialize-socket.ts:4]: Why do we use NX as the prefix ?

File: package.json [hunk 2]
```diff
@@ -27,6 +28,7 @@
     "@reduxjs/toolkit": "^1.8.6",
     "@rematch/core": "^2.2.0",
     "core-js": "^3.6.5",
+    "cors": "^2.8.5",
```
Comment (dalanad) [inline package.json]: Is this package a peer dependency?  I don't see it getting used

File: apps/server/src/main.ts [hunk 3]
```diff
@@ -1,11 +1,33 @@
...
   res.send('Hello World!');
 });
 
-app.listen(port, () => {
-  console.log(`App listening on port ${port}`);
+const allowedOrigin =
+  process.env.NX_ALLOWED_ORIGINS.split(', ') || 'http://localhost:4200';
+
+const io = new Server(server, {
+  cors: {
```
Comment (supunTE) [inline apps/server/src/main.ts:16]: @dalanad In here I'm using `cors` package to configure the `Access-Control-Allow-Origin`

File: package.json (hunk 2 above)
Comment (supunTE) [reply package.json]: Mentioned in, 
https://github.com/the-ai-team/razor/pull/53/files#r1119848113

File: apps/client/src/services/initialize-socket.ts (hunk 1 above)
Comment (supunTE) [reply apps/client/src/services/initialize-socket.ts:4]: Nx checks for the`NX` prefix to add when building.

image
https://nx.dev/recipes/environment-variables/use-environment-variables-in-react

File: apps/server/src/main.ts (hunk 3 above)
Comment (dalanad) [reply apps/server/src/main.ts:16]: doesn't it automatically get installed with socket.io?

Comment (dalanad) [reply apps/server/src/main.ts:16]: @supunTE

Comment (supunTE) [reply apps/server/src/main.ts:16]: image
https://socket.io/docs/v3/handling-cors/

Comment (supunTE) [reply apps/server/src/main.ts:16]: removed the package d725ceb

Comment (supunTE) [reply apps/server/src/main.ts:16]: Yeah, we don't need cors to install manually. socket IO is providing us with that. I misunderstood the issue.
removed the package https://github.com/the-ai-team/razor/commit/d725ceb42b18d42d53f02fb0d09390fcb0cea0ae

**Review by dalanad — APPROVED** [review]

**Pull request merged**